src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -160,13 +190,49 @@ export const Deposit: React.FC<DepositProps> = ({
   const toast = useToast()
 
   const supportedEvmChainIds = useMemo(() => getSupportedEvmChainIds(), [])
+
+  useEffect(() => {
+    if (!inputValues || !accountId)
+      return // Router contract address is only set in case we're depositting a token, not a native asset
+    ;(async () => {
+      const isApprovalRequired = await (async () => {
```

Comment:
theres a hook `useIsApprovalNeeded` which is trade quote specific, but it would be fairly easy to split up to generalise here, although i suspect it may be a lot of effort for little gain

Reply:
Yeah I was originally trying to reuse it and found out it has trade-specific signature. Would definitely keep the current implementation for now but happy to generalize `useIsApprovalNeeded` as a direct follow-up!

Reply:
that would be mega cool 🤝